dev/ToastNotifications/ToastActivationInfo.cpp

```diff
@@ -0,0 +1,38 @@
+﻿#include "pch.h"
+#include "ToastActivationInfo.h"
+#include "Microsoft.Windows.ToastNotifications.ToastActivationInfo.g.cpp"
+
+namespace winrt::Microsoft::Windows::ToastNotifications::implementation
+{
+    winrt::Microsoft::Windows::ToastNotifications::ToastActivationInfo ToastActivationInfo::CreateFromActivationGuid(winrt::guid const& taskClsid)
+    {
+        THROW_HR_IF_MSG(E_ILLEGAL_METHOD_CALL, !AppModel::Identity::IsPackagedProcess(), "Not applicable for unpackaged applications");
+
+        THROW_HR_IF(E_INVALIDARG, (taskClsid == winrt::guid(GUID_NULL)));
+
+        return winrt::make<ToastActivationInfo>(taskClsid);
+    }
+
+    winrt::Microsoft::Windows::ToastNotifications::ToastActivationInfo ToastActivationInfo::CreateFromToastAssets(winrt::Microsoft::Windows::ToastNotifications::ToastAssets const& assets)
+    {
+        THROW_HR_IF_MSG(E_ILLEGAL_METHOD_CALL, AppModel::Identity::IsPackagedProcess(), "Not applicable for packaged applications");
+
+        THROW_HR_IF(E_INVALIDARG, (assets == nullptr));
```

Comment:
THROW_HR_IF_NULL(E_INVALIDARG, assets)

Reply:
Although strictly speaking, can this actually be nullptr? Its a pass by reference right?

Reply:
@hulumane another design question here: What if the assets is not nullptr but if the asset properties i.e. the displayname or the iconuri is a nullptr should we throw pavan?

Reply:
If IconUri is blank, shell uses a default icon. I'm not sure about the DisplayName case.

Reply:
Discussed this offline, we are going to throw if the developer fails to provide display name or the iconuri. Thanks.